viewing_party/party.py

```diff
@@ -1,23 +1,183 @@
 # ------------- WAVE 1 --------------------
 
-def create_movie(title, genre, rating):
-    pass
+def create_movie(movie_title, genre, rating):
+    if movie_title == None or genre == None or rating == None:
+        return None
+    movie = {"title":movie_title, "genre":genre, "rating":rating}
+    return movie
 
+def add_to_watched(user_data, movie):
+    if movie["title"] not in user_data["watched"]:
+        user_data["watched"].append(movie)
+    return user_data
+    
+def add_to_watchlist(user_data, movie):
+    if movie["title"] not in user_data["watchlist"]:
+        user_data["watchlist"].append(movie)
+    return user_data
+    
+def watch_movie(user_data, movie_title):
+    for movie in user_data["watchlist"]:
+        if movie_title == movie["title"] and movie_title not in user_data["watched"]:
+            user_data["watchlist"].remove(movie)
+            user_data["watched"].append(movie)

```

Comment:
❗️We should be very careful about modifying a collection that we're currently iterating over. It ends up not mattering here, since our examples expect only a single value to be removed, so any odd iteration behavior after that doesn't cause a problem (note that some languages will intentionally crash when they detect that a modification has occurred to a collection, and an iteration over the collection continues).

To make it very clear that we are following safe behaviors, we should either:

1. Terminate the loop as soon as we have completed the modification (for example, adding `break` to line 30), or

2. Only use the loop to find the value we want to remove (either the entire value, if we later use `remove`, or the position, if we later use `pop`), and then remove the value after the loop completes.
 
Here's a [resource](https://support.techsmart.codes/hc/en-us/articles/360058068873-For-Loops-Why-shouldn-t-I-add-to-or-remove-from-a-list-during-a-for-each-loop) about why we shouldn't mutate a collection while iterating over it for you to review.

```suggestion
            user_data["watched"].append(movie)
            break
```